src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -317,6 +335,254 @@ def on_time_step__prepare(self, event: Event):
+        child_rp_with_parent = household_structure.loc[
+            child_rp_idx.intersection(parents_of_rp_idx)
+        ]
+        if len(child_rp_with_parent) > 0:
+            child_rp_with_parent_ids = (
+                child_rp_with_parent.reset_index()
+                .groupby(["child_id"])["person_id"]
+                .apply(np.random.choice)
+            )
+            pop.loc[child_rp_with_parent_ids.index, "guardian_1"] = child_rp_with_parent_ids
+
+        # Child is reference person with no parent, assign to age bound relative - blue box
+        child_rp_with_relative_ids = household_structure.loc[
+            child_rp_idx.drop(child_rp_with_parent.index.unique("child_id"), level="child_id")
+            .intersection(relatives_of_rp_idx)
+            .intersection(age_bound_idx)
+        ]
+        if len(child_rp_with_relative_ids) > 0:
+            child_rp_with_relative_ids = (
+                child_rp_with_relative_ids.reset_index()
+                .groupby(["child_id"])["person_id"]
+                .apply(np.random.choice)
+            )
+            pop.loc[
+                child_rp_with_relative_ids.index, "guardian_1"
+            ] = child_rp_with_relative_ids
+
+        # Child is not related to and is not reference person, assign to age bound non-relative - blurple box
+        non_relative_guardian = household_structure.loc[
+            child_non_relative_of_rp_idx.intersection(non_relatives_of_rp_idx).intersection(
+                age_bound_idx
+            )
+        ]
+        if len(non_relative_guardian) > 0:
+            non_relative_guardian_ids = (
+                non_relative_guardian.reset_index()
+                .groupby(["child_id"])["person_id"]
+                .apply(np.random.choice)
+            )
+            pop.loc[non_relative_guardian_ids.index, "guardian_1"] = non_relative_guardian_ids
+
+        # Child is not reference person, no age bound non-relative but there is adult non-relative - purple box
+        other_non_relative_guardian_ids = household_structure.loc[
+            child_non_relative_of_rp_idx.drop(
+                non_relative_guardian.index.unique("child_id"), level="child_id"
+            )
+            .intersection(non_relatives_of_rp_idx)
+            .intersection(over_18_idx)
+        ]
+        if len(other_non_relative_guardian_ids) > 0:
+            other_non_relative_guardian_ids = (
+                other_non_relative_guardian_ids.reset_index()
+                .groupby(["child_id"])["person_id"]
+                .apply(np.random.choice)
+            )
+            pop.loc[
+                other_non_relative_guardian_ids.index, "guardian_1"
+            ] = other_non_relative_guardian_ids
+
+        return pop
+
+    @staticmethod
+    def get_household_structure(pop: pd.DataFrame) -> pd.DataFrame:
+        """
+
+        Parameters
+        ----------
+        pop: population state table
+
+        Returns
```

Comment:
can you add slightly more explanation to this? i think i understand it after looking at it for a while, but it took a bit. my understanding is:

for every household X containing a simulant under 18 years old, there will be N rows in the returned table corresponding to household X

N = (number of sub-18 year olds in household X) * (number of people in household X - 1)

then, each row of the table will denote a comparison between one such sub-18 year old and someone else in their household. the included columns are the two sims' ages, relationships to head of household, and their age difference

is that correct?

Reply:
^ something to this effect would have helped me comprehend

Reply:
and / or, perhaps renaming "age_x" and "age_y" to "child_age" and "comparison_age" or something like that would have made it more readable without writing the kind of thing i just wrote

Reply:
TRhis is correct but I can update the doc string and the column names.